util/dvsim/FpvCfg.py

```diff
@@ -0,0 +1,243 @@
+                ])
+                summary += fpv_summary["pass_rate"]
+                if len(table) > 1:
+                    results_str += tabulate(table, headers="firstrow", tablefmt="pipe",
+                                            colalign=colalign)
+                else:
+                    results_str += "No fpv_summary found \n"
+                    summary += "N/A"
+            else:
+                results_str += "No fpv_summary found \n"
+                summary += "N/A"
+
+            if (self.cov):
+                cov_header = ["stimuli", "coi", "proof"]
+                cov_table = [cov_header]
+                cov_colalign = ("center", ) * len(cov_header)
+                results_str = "\n\n## Coverage Results\n"
+                results_str += ("### Coverage html file dir: " +
+                                self.scratch_path + "/default/formal-icarus \n\n")
+
+                if "fpv_coverage" in self.result:
+                    fpv_coverage = self.result["fpv_coverage"]
+                    cov_table.append([
+                        fpv_coverage["stimuli"],
+                        fpv_coverage["coi"],
+                        fpv_coverage["proof"]
+                    ])
+                    summary += fpv_coverage["stimuli"]
+                    summary += fpv_coverage["coi"]
+                    summary += fpv_coverage["proof"]
+
+                    results_str += "\n\n## Coverage Results\n"
+                    results_str += ("### Coverage html file dir: " +
+                                    self.scratch_path + "/default/formal-icarus \n\n")
+                    if len(cov_table) > 1:
+                        self.results_str += tabulate(cov_table, headers="firstrow",
+                                                     tablefmt="pipe", colalign=cov_colalign)
+
+                    else:
+                        results_str += "No fpv_coverage found \n"
+                        summary += ["N/A", "N/A", "N/A"]
+                else:
+                    results_str += "No fpv_coverage found \n"
+                    summary += ["N/A", "N/A", "N/A"]
+
+            if "messages" in self.result:
+                results_str += self.parse_dict_to_str(self.result["messages"])
+
+        # Write results to the scratch area
+        self.results_md = results_str
+        self.results_file = self.scratch_path + "/results_" + self.timestamp + ".md"
+        with open(self.results_file, 'w') as f:
+            f.write(self.results_md)
+
+        log.info("[results page]: [%s] [%s]", self.name, self.results_file)
+
+        # Generate result summary
+        if not self.cov:
+            summary += ["N/A", "N/A", "N/A"]
+        self.result_summary = summary
+
+        return self.results_md
+
+    def _publish_results(self):
+        '''Upon agreement with tool vendors, this script will only publish the result summary to
+        the public domain.
+        Individual detailed results/numbers will not be uploaded.
```

Comment:
I'm not sure this follows the usual Python docstring formatting. Also, I initially read this as meaning that the function published something to the public domain, and not anywhere else. Maybe something like this?

```python
        '''This does nothing: detailed messages from FPV runs are not published

        Our agreement with tool vendors allows us to publish the summary
        results (as in gen_results_summary), but not anything more detailed.

        '''
```